Duplicating a path when no path is selected causes NPE

**Describe the bug**  
When the duplicate button is pressed when no path is selected, a NullPointerException appears in the console.

Relevant lines of error:
`Caused by: java.lang.NullPointerException
	at edu.wpi.first.pathui.PathDisplayController.flip(PathDisplayController.java:285)
	at edu.wpi.first.pathui.MainController.flipVertical(MainController.java:238)
	... 66 more`

--- a/src/main/java/edu/wpi/first/pathweaver/MainController.java
+++ b/src/main/java/edu/wpi/first/pathweaver/MainController.java
@@ -59,9 +59,9 @@ public class MainController {
     paths.setEditable(true);
     setupEditable();
 
-    duplicate.disableProperty().bind(pathDisplayController.currentPath.isNull());
-    flipHorizontal.disableProperty().bind(pathDisplayController.currentPath.isNull());
-    flipVertical.disableProperty().bind(pathDisplayController.currentPath.isNull());
+    duplicate.disableProperty().bind(pathDisplayController.currentPathProperty().isNull());
+    flipHorizontal.disableProperty().bind(pathDisplayController.currentPathProperty().isNull());
+    flipVertical.disableProperty().bind(pathDisplayController.currentPathProperty().isNull());
 
   }
 

--- a/src/main/java/edu/wpi/first/pathweaver/MainController.java
+++ b/src/main/java/edu/wpi/first/pathweaver/MainController.java
@@ -32,6 +32,8 @@ public class MainController {
   private TreeItem<String> selected = null;
 
   @FXML private Button duplicate;
+  @FXML private Button flipHorizontal;
+  @FXML private Button flipVertical;
 
   @FXML
   private void initialize() {
@@ -58,6 +60,9 @@ public class MainController {
     setupEditable();
 
     duplicate.disableProperty().bind(pathDisplayController.currentPath.isNull());
+    flipHorizontal.disableProperty().bind(pathDisplayController.currentPath.isNull());
+    flipVertical.disableProperty().bind(pathDisplayController.currentPath.isNull());
+
   }
 
   @SuppressWarnings("PMD.NcssCount")

--- a/src/main/java/edu/wpi/first/pathweaver/MainController.java
+++ b/src/main/java/edu/wpi/first/pathweaver/MainController.java
@@ -5,6 +5,7 @@ import java.util.List;
 
 import javafx.event.EventHandler;
 import javafx.fxml.FXML;
+import javafx.scene.control.Button;
 import javafx.scene.control.TreeItem;
 import javafx.scene.control.TreeView;
 import javafx.scene.input.KeyCode;
@@ -30,6 +31,8 @@ public class MainController {
 
   private TreeItem<String> selected = null;
 
+  @FXML private Button duplicate;
+
   @FXML
   private void initialize() {
     pathDirectory = directory + "Paths/";
@@ -53,6 +56,8 @@ public class MainController {
     autons.setEditable(true);
     paths.setEditable(true);
     setupEditable();
+
+    duplicate.disableProperty().bind(pathDisplayController.currentPath.isNull());
   }
 
   @SuppressWarnings("PMD.NcssCount")

--- a/src/main/java/edu/wpi/first/pathweaver/PathDisplayController.java
+++ b/src/main/java/edu/wpi/first/pathweaver/PathDisplayController.java
@@ -28,7 +28,8 @@ public class PathDisplayController {
   private Pane topPane;
   private final PseudoClass selected = PseudoClass.getPseudoClass("selected");
   private Waypoint selectedWaypoint = null;
-  public final ObjectProperty<Path> currentPath = new SimpleObjectProperty<>();
+
+  private final ObjectProperty<Path> currentPath = new SimpleObjectProperty<>();
   private final Field field = new Field();
 
   private final ObservableList<Path> pathList = FXCollections.observableArrayList();
@@ -282,29 +283,17 @@ public class PathDisplayController {
     currentPath.get().flip(horizontal, drawPane);
   }
 
-  /**
-   * Retrieves a named Path.
-   *
-   * @param name The name of the Path to retrieve
-   *
-   * @return The appropriate Path from the Path List
-   */
-  public Path getPath(String name) {
-    for (Path p : pathList) {
-      if (p.getPathName().equals(name)) {
-        return p;
-      }
-    }
-    // TODO: Return a default path
-    return null;
-  }
-
   public void setPathDirectory(String pathDirectory) {
     this.pathDirectory = pathDirectory;
   }
 
   public String getPathDirectory() {
     return pathDirectory;
+  }
+
+
+  public ObjectProperty<Path> currentPathProperty() {
+    return currentPath;
   }
 
   /**

--- a/src/main/java/edu/wpi/first/pathweaver/PathDisplayController.java
+++ b/src/main/java/edu/wpi/first/pathweaver/PathDisplayController.java
@@ -28,7 +28,7 @@ public class PathDisplayController {
   private Pane topPane;
   private final PseudoClass selected = PseudoClass.getPseudoClass("selected");
   private Waypoint selectedWaypoint = null;
-  private final ObjectProperty<Path> currentPath = new SimpleObjectProperty<>();
+  public final ObjectProperty<Path> currentPath = new SimpleObjectProperty<>();
   private final Field field = new Field();
 
   private final ObservableList<Path> pathList = FXCollections.observableArrayList();

--- a/src/main/java/edu/wpi/first/pathweaver/PathDisplayController.java
+++ b/src/main/java/edu/wpi/first/pathweaver/PathDisplayController.java
@@ -17,6 +17,7 @@ import javafx.scene.input.KeyEvent;
 import javafx.scene.layout.Pane;
 import javafx.scene.transform.Scale;
 
+@SuppressWarnings("PMD.TooManyMethods")
 public class PathDisplayController {
   @FXML
   private ImageView backgroundImage;

